lib/data-types.js

```diff
@@ -481,6 +534,33 @@ DATEONLY.prototype._stringify = function _stringify(date) {
 };
 
 /**
+ * A YEAR column.
+ *
+ *
+ * @property YEAR
+ */
+function YEAR() {
+  if (!(this instanceof YEAR)) return new YEAR();
+}
+util.inherits(YEAR, ABSTRACT);
+
+YEAR.prototype.key = YEAR.key = 'YEAR';
+YEAR.prototype.toSql = function() {
+  return 'YEAR';
+};
+
+YEAR.prototype._stringify = function _stringify(date) {
+  return date;
+};
+
+YEAR.prototype.validate = function(value){
+  if (!Validator.isInt(String(value) || String(value).length !== 4 || String(value).length !== 2)){
```

Comment:
You are calling `Validator.isInt()` on the boolean expression `String(value) || String(value).length !== 4 || String(value).length !== 2`